javaee/impl/src/main/java/org/jboss/forge/addon/javaee/jpa/ui/JPANewEntityCommandImpl.java

```diff
@@ -74,7 +86,17 @@ public void initializeUI(UIBuilder builder) throws Exception
    {
       super.initializeUI(builder);
       idStrategy.setDefaultValue(GenerationType.AUTO);
-      builder.add(idStrategy).add(tableName);
+      idType.setDefaultValue(EntityIdType.LONG_PROPERTY);
+      idClass.setRequired(new Callable<Boolean>(){
```

Comment:
You can use a lambda here:`idClass.setRequired(() -> idType.getValue() == EntityIdType.EMBEDDED_ID);`

Reply:
:+1: